src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -210,6 +216,17 @@ export default class PhotoField extends React.Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    const nextFormData = nextProps.formData || {};
+    const prevFormData = this.props.formData || {};
+    if (nextFormData.file instanceof Blob && nextFormData.file !== prevFormData.file) {
+      if (this.state.previewSrc) {
+        window.URL.revokeObjectURL(this.state.previewSrc);
+      }
+      this.setState({ previewSrc: window.URL.createObjectURL(nextFormData.file) });
```

Comment:
Is there an advantage of using a this over the file itself as the preview source?

Reply:
As far as I could find, this is what you need to do to use the file itself as the preview source.